admin/release.py

```diff
@@ -934,6 +937,90 @@ def create_release_branch_main(args, base_path, top_level):
         raise SystemExit(1)
 
 
+class TestRedirectsOptions(Options):
+    """
+    Arguments for ``test-redirects`` script.
+    """
+    optParameters = [
+        ["doc-version", None, None,
+         "The version which the documentation sites are expected to redirect "
+         "to.\n"
+        ],
+    ]
+
+    optFlags = [
+        ["production", None, "Check the production documentation site."],
+    ]
+
+    environment = Environments.STAGING
+
+    def parseArgs(self):
+        if self['doc-version'] is None:
+            self['doc-version'] = get_doc_version(flocker.__version__)
+
+        if self['production']:
+            self.environment = Environments.PRODUCTION
+
+
+def test_redirects_main(args, base_path, top_level):
+    """
+    Tests redirects to Flocker documentation.
+
+    :param list args: The arguments passed to the script.
+    :param FilePath base_path: The executable being run.
+    :param FilePath top_level: The top-level of the flocker repository.
+    """
+    options = TestRedirectsOptions()
+
+    try:
+        options.parseOptions(args)
+    except UsageError as e:
+        sys.stderr.write("%s: %s\n" % (base_path.basename(), e))
+        raise SystemExit(1)
+
+    doc_version = options['doc-version']
+
+    document_configuration = DOCUMENTATION_CONFIGURATIONS[options.environment]
+    base_url = 'https://' + document_configuration.cloudfront_cname
+
+    is_dev = not is_release(doc_version)
+    if is_dev:
+        expected_redirects = {
+            '/en/devel': '/en/' + doc_version + '/',
+            '/en/devel/authors.html': '/en/' + doc_version + '/authors.html',
+        }
+    else:
+        expected_redirects = {
+            '/': '/en/' + doc_version + '/',
+            '/en/': '/en/' + doc_version + '/',
+            '/en/latest': '/en/' + doc_version + '/',
+            '/en/latest/authors.html': '/en/' + doc_version + '/authors.html',
+        }
+
+    failed_redirects = []
+
+    for path in expected_redirects:
+        original_url = base_url + path
+        expected_url = base_url + expected_redirects[path]
+        final_url = requests.get(original_url).url
+
+        if expected_url != final_url:
+            failed_redirects.append(original_url)
+
+            message = (
+                "'{original_url}' expected to redirect to '{expected_url}', "
+                "instead redirects to '{final_url}'.\n").format(
+                    original_url=original_url,
+                    expected_url=expected_url,
+                    final_url=final_url,
+            )
+
+            sys.stderr.write(message)
```

Comment:
Add a message if all the redirects pass, to confirm that the test passed successfully.